### ISSUE TITLE
The Cygwin version of cntlm fails pthread_join()

### ISSUE THREAD

**Issue by fralken** [versat/cntlm#49]
I'm testing the Windows version of cntlm built with Cygwin.
It works, but each "pthread_join()" fails with the following log:

`Serious error during pthread_join: 22`

22 corresponds to `EINVAL` error.
I built it with Cygwin version 3.3.4 with the default configuration (no kerberos and no pacparser).
This is reproducible by running cntlm from a _cmd_ terminal with `-f` parameter to keep it in foreground.

The interesting thing is that this doesn't happen when launched from the Cygwin's shell. In this case pthread_join() doesn't fail. 
Can this be related to issue #18?

**Comment by versat** [comment]
I have heavily used the cntlm Cygwin version and debugged it. While I have found some issues in cntlm (and I guess there still are some), it seemed to me that Cygwin is not that stable.
Sadly Cygwin is a bit "chaotic". Sources are not that easy to build and documentation really could be improved. So I did not debug into Cygwin or looked  deeply into the sources.
It is only my impression, though.

Interesting that it works in the Cygwin shell but never outside of it. I have no idea what could be wrong there.

**Comment by fralken** [comment]
After further investigation I found that this problem affects only one pc. I tested on another Windows machine and it works flawlessly.
The affected pc is a corporate machine that is full of antivirus and hooks, and it is reasonable to think that there can be hooks that are triggered by programs executed from the cmd shell, while they are not triggered when launched from the Cygwin bash, and they conflicts in some way with the Cygwin implementation of pthreads.

Also I did some stress tests and started hundreds of connections concurrently without problems, so I guess that the error is a false positive, even if a bit annoying.

Nevertheless I noticed that maybe pthread usage in cntlm is not properly implemented for these reasons:

1. threads do not return a value, so there is no need for the main thread to join them. They could be started as _detached_ threads instead.
2. socks5_thread is not joined, because its id is not put in the threads_list to be joined later
3. the _serialize_ flag is only used for proxy_thread, while it should be used also for socks5_thread and tunnel_thread (if there is the need to debug cntlm in single thread mode)

I will do some further tests on this and possibly prepare a new PR.
